src/python/turicreate/toolkits/drawing_classifier/_tf_drawing_classifier.py

```diff
@@ -85,42 +94,57 @@ def __init__(self, net_params, batch_size, num_classes):
 
         # Loss
         self.cost = _tf.reduce_mean(_tf.nn.softmax_cross_entropy_with_logits_v2(logits=self.predictions,
-            labels=self.one_hot_labels))
+                                                                                labels=self.one_hot_labels))
 
         # Optimizer
-        self.optimizer = _tf.train.AdamOptimizer(learning_rate=0.001).minimize(self.cost)
+        self.optimizer = _tf.train.AdamOptimizer(
+            learning_rate=0.001).minimize(self.cost)
 
         # Predictions
-        correct_prediction = _tf.equal(_tf.argmax(self.predictions, 1), _tf.argmax(self.one_hot_labels, 1))
-        self.accuracy = _tf.reduce_mean(_tf.cast(correct_prediction, _tf.float32))
-        
+        correct_prediction = _tf.equal(_tf.argmax(
+            self.predictions, 1), _tf.argmax(self.one_hot_labels, 1))
+        self.accuracy = _tf.reduce_mean(
+            _tf.cast(correct_prediction, _tf.float32))
+
         self.sess = _tf.Session()
         self.sess.run(_tf.global_variables_initializer())
 
         # Assign the initialised weights from MXNet to tensorflow
         layers = ['drawing_conv0_weight', 'drawing_conv0_bias', 'drawing_conv1_weight', 'drawing_conv1_bias',
-        'drawing_conv2_weight', 'drawing_conv2_bias', 'drawing_dense0_weight', 'drawing_dense0_bias',
-        'drawing_dense1_weight', 'drawing_dense1_bias']
+                  'drawing_conv2_weight', 'drawing_conv2_bias', 'drawing_dense0_weight', 'drawing_dense0_bias',
+                  'drawing_dense1_weight', 'drawing_dense1_bias']
 
         for key in layers:
             if 'bias' in key:
                 self.sess.run(_tf.assign(_tf.get_default_graph().get_tensor_by_name(key+":0"),
-                    net_params[key]))
+                                         net_params[key]))
             else:
-                if 'dense' in key:
-                    dense_weights = _utils.convert_dense_coreml_to_tf(net_params[key])
+                if 'drawing_dense0_weight' in key:
+                    '''
+                    To make output of MXNET pool3 (NCHW) compatible with TF (NHWC).
+                    Decompose FC weights to NCHW. Transpose to NHWC. Reshape back to FC.
+                    '''
+                    mxnet_128_576 = net_params[key]
+                    mxnet_128_576 = _np.reshape(mxnet_128_576, (128, 64, 3, 3))
+                    mxnet_128_576 = _np.transpose(mxnet_128_576, (0, 2, 3, 1))
+                    mxnet_128_576 = _np.reshape(mxnet_128_576, (128, 576))
```

Comment:
It looks like this logic was removed in #2514. @shantanuchhabra Do you have context on why? Just want to make sure we're not waffling back and forth between two issues

Reply:
This is about the warm start part, which reads the pre-trained weights from our side.

Reply:
Looks like removing the code was an oversight on my part. In an effort to reduce code and simplify, I didn't realize that `dense0` was a special case that the conversion utilities in `_tf_utils.py` do not take care of. Apologies!

Reply:
Sorry, I missed this during code review too, @fareeha.

When in doubt, please err on the side of understanding/asking about the code you're deleting without replacing, especially when the original author went through the trouble of documenting it as a special case! We lost some development time due to re-engineering this.